app/code/Magento/CatalogImportExport/Model/Import/Product/Validator/Weight.php

```diff
@@ -10,23 +10,25 @@
 class Weight extends AbstractImportValidator implements RowValidatorInterface
 {
     /**
-     * {@inheritdoc}
+     * @inheritdoc
      */
     public function isValid($value)
     {
         $this->_clearMessages();
         if (!empty($value['weight']) && (!is_numeric($value['weight']) || $value['weight'] < 0)
             && $value['weight'] !== $this->context->getEmptyAttributeValueConstant()
         ) {
-            $this->_addMessages(
-                [
-                    sprintf(
-                        $this->context->retrieveMessageTemplate(self::ERROR_INVALID_ATTRIBUTE_TYPE),
-                        'weight',
-                        'decimal'
-                    )
-                ]
-            );
+            if ($this->context->retrieveMessageTemplate(self::ERROR_INVALID_ATTRIBUTE_TYPE) !== null) {
```

Comment:
This slightly changes method logic. In this case I would fix test itself.
I believe we can allow call to original method in this case:
https://github.com/magento/magento2/blob/203a44f9e755fa6d2e057f1b99efbaff17546a80/app/code/Magento/CatalogImportExport/Test/Unit/Model/Import/Product/Validator/WeightTest.php#L33

Reply:
as it's a mock, and message doesn't have an impact on the validation result in the test, I've added a mock value to that line:33
please re-check